### ISSUE TITLE
Allow drawer to use focus trap

### ISSUE THREAD

**Issue by redallen** [patternfly/patternfly-react#5354]
Add option for drawer to use focus trap so focus can return to the previous element when you close the drawer.

**Comment by jessiehuff** [comment]
Are there any downsides to using focus trap here?

**Comment by redallen** [comment]
Technically no. Experience-wise I think Drawers can be used for many different things and you might not expect a user to be able to focus a drawer at all.

**Comment by kdoberst** [comment]
Wondering if we can get an update on the priority of this bug.  We want to implement a drawer but by not having a focus trap, the user can put focus on items that are behind the drawer which is an accessibility violation (see https://www.w3.org/WAI/WCAG21/Understanding/focus-visible.html)
